src/integrations/prefect-azure/prefect_azure/blob_storage.py

```diff
@@ -119,16 +124,76 @@ def example_blob_storage_upload_flow():
     logger = get_run_logger()
     logger.info("Uploading blob to container %s with key %s", container, blob)
 
-    # create key if not provided
+    # Create key if not provided
     if blob is None:
         blob = str(uuid.uuid4())
 
     async with blob_storage_credentials.get_blob_client(container, blob) as blob_client:
-        await blob_client.upload_blob(data, overwrite=overwrite)
+        try:
+            await blob_client.upload_blob(data, overwrite=overwrite)
+        except ResourceNotFoundError as exc:
+            logger.error("Error uploading blob %s to container %s: %s", blob, container, exc.reason)
+            raise RuntimeError(f"Upload failed: {exc.reason}") from exc
 
     return blob
 
 
+@task
```

Comment:
this [appears](https://github.com/PrefectHQ/prefect/actions/runs/10945006452/job/30388213912?pr=15435#step:8:37) to be a duplicate of the below function?